src/vs/workbench/contrib/terminal/browser/xterm/xtermTerminal.ts

```diff
@@ -368,20 +368,49 @@ export class XtermTerminal extends Disposable implements IXtermTerminal, IDetach
 		return this._serializeAddon.serializeAsHTML();
 	}
 
-	async getSelectionAsHtml(command?: ITerminalCommand): Promise<string> {
+	async getSelectionAsHtml(command?: ITerminalCommand, maxLines?: number, explicitRange?: boolean): Promise<string> {
 		if (!this._serializeAddon) {
 			const Addon = await this._xtermAddonLoader.importAddon('serialize');
 			this._serializeAddon = new Addon();
 			this.raw.loadAddon(this._serializeAddon);
 		}
-		if (command) {
+		if (explicitRange && command) {
+			let startLine: number;
+			let startCol: number;
+			if (command.executedMarker && command.executedMarker.line >= 0) {
+				startLine = command.executedMarker.line;
+				startCol = Math.max(command.executedX ?? 0, 0);
+			} else {
+				startLine = command.marker?.line !== undefined ? command.marker.line + 1 : 1;
+				startCol = Math.max(command.startX ?? 0, 0);
+			}
+
+			const endLine = command.endMarker?.line !== undefined ? command.endMarker.line - 1 : this.raw.buffer.active.length - 1;
+			if (endLine < startLine) {
+				return '';
+			}
+			if (maxLines && endLine - startLine > maxLines) {
+				startLine = endLine - maxLines;
+				startCol = 0;
+			}
+
+			const bufferLine = this.raw.buffer.active.getLine(startLine);
+			if (bufferLine) {
+				startCol = Math.min(startCol, bufferLine.length);
+			}
+
+			const range = { startLine, endLine, startCol };
+			const result = this._serializeAddon.serializeAsHTML({ range });
+			return result;
```

Comment:
We may want to massage the start/end lines to trim empty lines or maybe just 1 empty line on either side?